components/job-orchestration/job_orchestration/scheduler/compress/compression_scheduler.py

```diff
@@ -0,0 +1,319 @@
+import argparse
+import celery
+import datetime
+import logging
+import os
+import sys
+import time
+from contextlib import closing
+from pathlib import Path
+
+import msgpack
+import zstandard
+from pydantic import ValidationError
+
+from clp_package_utils.general import CONTAINER_INPUT_LOGS_ROOT_DIR
+from clp_py_utils.clp_config import (
+    CLPConfig,
+    COMPRESSION_JOBS_TABLE_NAME,
+    COMPRESSION_TASKS_TABLE_NAME
+)
+from clp_py_utils.clp_logging import get_logger, get_logging_formatter, set_logging_level
+from clp_py_utils.compression import validate_path_and_get_info
+from clp_py_utils.core import read_yaml_config_file
+from clp_py_utils.sql_adapter import SQL_Adapter
+from job_orchestration.executor.compress.fs_compression_task import compress
+from job_orchestration.scheduler.constants import (
+    CompressionJobStatus,
+    CompressionTaskStatus
+)
+from job_orchestration.scheduler.compress.partition import PathsToCompressBuffer
+from job_orchestration.scheduler.job_config import ClpIoConfig
+from job_orchestration.scheduler.scheduler_data import (
+    CompressionJob,
+    CompressionTaskSuccessResult,
+    CompressionTaskFailureResult,
+)
+
+# Setup logging
+logger = get_logger(__name__)
+
+scheduled_jobs = {}
+
+
+def fetch_new_jobs(db_cursor):
+    db_cursor.execute(f"""
+        SELECT id, clp_config, creation_time
+        FROM {COMPRESSION_JOBS_TABLE_NAME}
+        WHERE status='{CompressionJobStatus.PENDING}'
+    """)
+    return db_cursor.fetchall()
+
+
+def update_compression_task_metadata(db_cursor, task_id, kv):
+    if not len(kv):
+        logger.error("Must specify at least one field to update")
+        raise ValueError
+
+    field_set_expressions = [f'{k}="{v}"' for k, v in kv.items()]
+    query = f'UPDATE {COMPRESSION_TASKS_TABLE_NAME} SET {", ".join(field_set_expressions)} WHERE id={task_id};'
+    db_cursor.execute(query)
+
+
+def update_compression_job_metadata(db_cursor, job_id, kv):
+    if not len(kv):
+        logger.error("Must specify at least one field to update")
+        raise ValueError
+
+    field_set_expressions = [f'{k}="{v}"' for k, v in kv.items()]
+    query = f'UPDATE {COMPRESSION_JOBS_TABLE_NAME} SET {", ".join(field_set_expressions)} WHERE id={job_id};'
+    db_cursor.execute(query)
+
+
+def search_and_schedule_new_tasks(db_conn, db_cursor, clp_metadata_db_connection_config):
+    """
+    For all jobs with PENDING status, split the job into tasks and schedule them.
+    """
+    global scheduled_jobs
+
+    logger.debug('Search and schedule new tasks')
+
+    zstd_dctx = zstandard.ZstdDecompressor()
+    zstd_cctx = zstandard.ZstdCompressor(level=3)
+
+    # Poll for new compression jobs
+    jobs = fetch_new_jobs(db_cursor)
+    db_conn.commit()
+    for job_row in jobs:
```

Comment:
Can we move `db_conn.commit()` into `fetch_new_jobs` (to keep it modular if we ever use the function elsewhere) and then we can simplify the loop to:

```suggestion
    for job_row in fetch_new_jobs(db_conn, db_cursor):
```

Reply:
But we need to pass `db_conn` to` fetch_new_jobs` while other database functions don't require this parameter and all of them commit outside the function

Reply:
Fair enough. I think the database ops need some refactoring to be less error prone but we can do that in another PR.